src/server/routes/signin.js

```diff
@@ -21,15 +21,15 @@ export default async function signin(req, res) {
     return res.status(500).end(`Error: Type not specified for ${provider.name}`)
   }
 
-  if (provider.type === "oauth" && req.method === "POST") {
+  if (provider.type === "oauth") {
     try {
       const authorizationUrl = await getAuthorizationUrl(req)
       return res.redirect(authorizationUrl)
     } catch (error) {
       logger.error("SIGNIN_OAUTH_ERROR", error)
       return res.redirect(`${baseUrl}${basePath}/error?error=OAuthSignin`)
     }
-  } else if (provider.type === "email" && req.method === "POST") {
+  } else if (provider.type === "email") {
```

Comment:
I think these changes may introduce a vulnerability by letting the user visit URLs directly from within their browsers. POST requests mitigate this issue.

Reply:
So the reason I think it's not a problem is that we already check if the req is a `POST` here:

https://github.com/nextauthjs/next-auth/blob/613c3033156652b1a93ba3376f99c7ee3267cdb2/src/server/index.js#L230-L237

This is similar to https://github.com/nextauthjs/next-auth/blob/main/src/server/routes/callback.js which will only run with `POST` but doesn't do this check either.

In the long term, the less references we have to `req` outside of `src/server/index.js`, the easier it will be to lift out these methods and make them pure, ultimately contributing to functionality like fetch-less `getSession`, etc.

Reply:
Oh, that’s right, thanks for the explanation! This looks good to me, then 😊